docs/transfer.md

```diff
@@ -0,0 +1,32 @@
+# Transfer Service
+
+The transfer service is a simple flexible service which can be used to transfer artifact objects between a source and destination. The service determines whether the transfer between the source and destination is possible rather than the API.
+
+The transfer service if built upon the core ideas put forth by the libchan project, that an API with binary streams and data channels as first class objects is more flexible and opens a wider variety of use cases without requiring constant protocol and API updates. To accomplish this, the transfer service makes use of the streaming service to allow binary and object streams to be accessible by transfer objects even when using grpc and ttrpc.
+
+
+## Transfer Objects (Sources and Destinations)
+
+## Transfer Operations
+
+|   Source    | Destination | Description |
+|-------------|-------------|-------------|
+| Registry    | Image Store | "pull"      |
+| Image Store | Registry    | "push"      |
+| Object stream (Archive) | Image Store | "import" |
+| Image Store | Object stream (Archive) | "export" |
+| Object stream (Layer) | Mount/Snapshot | "unpack" |
+| Mount/Snapshot | Object stream (Layer) | "diff" |
+| Image Store | Image Store | "tag" |
+| Registry | Registry | mirror registry image |
```

Comment:
When mirroring registry images, will it be a 2 step process that is abstracted 
```
registry1 -> image store
image store -> registry2
```

or a direct `registry -> registry` transfer

Reply:
The idea is that the transfer interface supports it, how it is implemented is up to the transfer plugin which implements it. The caller would just give two registry arguments though